services/crypto_delete_allowance.proto

```diff
@@ -29,34 +37,68 @@ option java_multiple_files = true;
 import "basic_types.proto";
 
 /**
- * Deletes one or more non-fungible approved allowances from an owner's account. This operation
- * will remove the allowances granted to one or more specific non-fungible token serial numbers. Each owner account
- * listed as wiping an allowance must sign the transaction. Hbar and fungible token allowances
- * can be removed by setting the amount to zero in CryptoApproveAllowance.
+ * Delete one or more allowances.<br/>
+ * Given one or more, previously approved, allowances for non-fungible/unique
+ * tokens to be transferred by a spending account from an owning account;
+ * this transaction removes a specified set of those allowances.
+ *
+ * The owner account for each listed allowance MUST sign this transaction.<br/>
+ * Allowances for HBAR cannot be removed with this transaction. The owner
+ * account MUST submit a new `cryptoApproveAllowance` transaction with the
+ * amount set to `0` to "remove" that allowance.<br/>
+ * Allowances for fungible/common tokens cannot be removed with this
+ * transaction. The owner account MUST submit a new `cryptoApproveAllowance`
+ * transaction with the amount set to `0` to "remove" that allowance.<br/>
+ *
+ * ### Block Stream Effects
+ * None
  */
 message CryptoDeleteAllowanceTransactionBody {
     /**
-     * List of non-fungible token allowances to remove.
+     * List of non-fungible/unique token allowances to remove.
+     * <p>
+     * This list MUST NOT be empty.
      */
     repeated NftRemoveAllowance nftAllowances = 2;
+
+    // Note: Field numbers 1 and 3 were very briefly used in early
+    //     versions of this file, but were removed prior to any release
+    //     version. Those field numbers MAY be reused.
 }
 
 /**
- * Nft allowances to be removed on an owner account
+ * A single allowance for one non-fungible/unique token.
+ * This is specific to removal, and the allowance is identified for that
+ * specific purpose.
+ *
+ * All fields in this message are REQUIRED.
+ * The `serial_numbers` list MUST NOT be empty.
+ * The combination of field values in this message MUST match existing
+ * allowances for one or more individual non-fungible/unique tokens.
+ *
+ * > REVIEW NOTE
+ * >> How does one remove an `approve_for_all` allowance?
```

Comment:
We need to use `CryptoApproveAllowanceHandler` with `approveForAll = false` to remove allowance

Reply:
I am not sure how that translates to transaction API.
Would this be a `CryptoApproveAllowanceTransactionBody` with a `NftAllowance` with `approved_for_all = false`?  How would that _remove_ the previously approved allowance?
We need to describe this carefully and thoroughly because it sounds extremely counter-intuitive (delete by creating), and seems like it's something dApp developers will find very confusing.

Reply:
Found an answer in the new TCK work; and incorporated that in the documentation here.